content/tutorials/spatial_point_in_polygon/index.Rmd

```diff
@@ -0,0 +1,188 @@
+
+```{r}
+crs_wgs84 <- CRS("+init=epsg:4326")
+coord <- invasive_species %>%
+    select(decimalLongitude, decimalLatitude)
+invasive_spatial <- SpatialPointsDataFrame(coord,
+                                           data = invasive_species,
+                                           proj4string = crs_wgs84)
+```
+
+We transform the 10 km x 10 km grid to the same CRS system:
+
+```{r}
+be10grid <- spTransform(be10grid, crs_wgs84)
+```
+
+Now we are ready to spatially join (overlay) the SpatialPointsDataFrame wth the 10 km x 10 km grid. This can be done using `sp::over`. The first two arguments of the function give, respectively, the geometry (locations) of the queries, and the layer from which the geometries or attributes are queried. See `?sp::over`. In this case, when x = "SpatialPoints" and  y = "SpatialPolygonsDataFrame", it
+returns a data.frame of the second argument with row entries corresponding to the first argument.
+
+```{r}
+invasive_be10grid <- over(x = invasive_spatial, y = be10grid)
+invasive_species_be10grid <- bind_cols(invasive_species,
+                                       invasive_be10grid)
+```
+
+To see what the result looks like, we can select the most relevant variables and print it (first ten rows).
+
+```{r}
+invasive_species_be10grid %>%
+  select(species, starts_with("decimal"),
+         eventDate, cellcode) %>%
+  head(10)
+```
+
+
+
+## Point in polygon with the `sf` package
+
+The same spatial query can be done with the aid of the `sf` package. The package has built-in functions to read spatial data (which also uses GDAL as backbone). 
+
+We will project the data to Lambert72, because the join assumes planar coordinates.
+
+```{r}
+be10grid <- read_sf(paste0(tempdir(), "\\Belgium.sqlite"), 
+                    layer = "be_10km")  %>%
+  # convert to Lambert72
+  st_transform(crs = 31370)
+```
+
+Instead of a SpatialPolygonsDataframe, we now get a "sf" object which is also a data.frame and a tbl:
+
+```{r}
+class(be10grid)
+```
+
+The spatial information now resides in a GEOMETRY variable:
+
+```{r}
+be10grid$GEOMETRY
+```
+
+
+Similarly, the package has built-in functions to convert a data.frame containing coordinates to a spatial data.frame:
+
+```{r}
+invasive_spatial <- st_as_sf(invasive_species,
+                             coords = c("decimalLongitude",
+                                        "decimalLatitude"),
+                             crs = 4326) %>%
+  # convert to Lambert72
+  st_transform(crs = 31370)
+```
+
+Now we are ready to make the spatial overlay. This is done with the aid of `sf::st_join`. We set the join type to `st_within` and join the information from the grid to the points through a left join. See [the DE-9IM topological model](https://en.wikipedia.org/wiki/DE-9IM) for explanations about all possible spatial joins. 
+
+
+```{r}
+invasive_be10grid <- invasive_spatial %>%
+  st_join(be10grid,
+          join = st_within, 
```

Comment:
I think the default setting `join = st_intersects` would do the job, no?

Reply:
Good point. But there is a difference. With ` sp::over`, points on a polygon boundary and points corresponding to a polygon vertex are considered to be inside the polygon. This corresponds with `st_intersects`, so I should have used that to get the same behavior. I will change this and note that `st_within` can be used if we want to join only when points are within (excluding boundary).

Reply:
OK, clear!